src/ci_providers/provider_buildkite.ts

```diff
@@ -85,27 +88,35 @@ function getServiceName() {
  * @param {args: {}, envs: {}} inputs an object of arguments and enviromental variable key/value pairs
  * @returns {string}
  */
-function _getSHA(inputs) {
+function _getSHA(inputs: UploaderInputs): string {
   const { args, envs } = inputs
-  return args.sha || envs.BUILDKITE_COMMIT
+  if (!!args.sha || !!envs.BUILDKITE_COMMIT) {
+    return args.sha || envs.BUILDKITE_COMMIT || ''
+  }
+  logAndThrow('Unable to detect sha, please set manually with the -C flag')
+  return ''
 }
```

Comment:
Although this function exists in the other provider implementations, this one seems to be written differently. While a future PR should centralize this behavior, the function as it exists should be updated to better correspond with that of the other providers.

Reply:
Hmm. They should all be the same. However, since this is part of the private implementation of https://github.com/codecov/uploader/blob/typescript-migration/src/types.ts#L32, does it matter how the writer implements it?

Reply:
It doesn't matter in terms of immediate functionality but rather for future maintenance. Functions that serve the same purpose can be discovered more easily if they are written similarly. Eventually, it would probably be ideal to merge their shared functionality:

- Less overall code footprint which means each line of code has more :eyes:
- Less code means less chance of functionality diverging from its working form and causing bugs (since, instead of just *one* place to review, there are *several*)